src/java/main/org/apache/zookeeper/server/auth/ProviderRegistry.java

```diff
@@ -87,4 +113,10 @@ public static String listProviders() {
         }
         return sb.toString();
     }
+
+    public static SortedSet<String> getSchemesRequiringAuth() {
```

Comment:
what if the user decided which schemes were needed rather than letting each individual scheme choose if it is essential?

Reply:
by 'user', i take it that you mean the client that is connecting/transacting with zookeeper (or is it the auth provider that you are referring to). typically, it is the server that decides what kind of security/auth policy needs to be in place for the client to connect/transact with it. if the clients can present those credentials, then we let it connect; else, reject it, if the auth provider *requires* the clients to authenticate with it before transacting. the auth provider may not have even been loaded to begin with, if we let the clients choose which schemes needs to be verified.